src/agent/rest_web_server.cpp

```diff
@@ -0,0 +1,1061 @@
+    Connection->header_f = new char[len+1]();
+   
+    
+    strncpy(Connection->header_f,at,len);
+    otbrLog(OTBR_LOG_ERR, "header_f: %s",Connection->header_f);
+    return 0;
+}
+
+static int on_header_value(http_parser *parser, const char *at, size_t len) {
+    Connection *connection = (struct Connection*)parser->data;
+    connection->header_v = new char[len+1]();
+    
+    strncpy(connection->header_v,at,len);
+    otbrLog(OTBR_LOG_ERR, "header_v: %s",connection->header_v);
+    return 0;
+}
+static int on_body(http_parser *parser, const char *at, size_t len) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->body = new char[len+1]();
+    strncpy(connection->body,at,len-1);
+    
+    otbrLog(OTBR_LOG_ERR, "body : %s",connection->body);
+    return 0;
+}
+
+static int on_headers_complete(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->content_length = parser->content_length;
+    connection->method = parser->method;
+    return 0;
+}
+
+static int on_message_complete(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->complete = 0;
+    return 0;
+}
+
+static int on_chunk_header(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->complete = 0;
+    return 0;
+}
+
+static int on_chunk_complete(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->complete = 0;
+    return 0;
+}
+
+static int setNonBlocking(int fd)
+{
+	int oldflags;
+
+	oldflags = fcntl(fd, F_GETFL);
+
+	if (fcntl(fd, F_SETFL, oldflags | O_NONBLOCK) == -1) {
+		return -1;
+	}
+
+	return 0;
+}
+
+static http_parser_settings mSettings {
+        .on_message_begin = on_message_begin,
+        .on_url = on_url,
+        .on_status = on_status,
+        .on_header_field = on_header_field,
+        .on_header_value = on_header_value,
+        .on_headers_complete = on_headers_complete,
+        .on_body = on_body,
+        .on_message_complete = on_message_complete,
+        .on_chunk_header = on_chunk_header,
+        .on_chunk_complete = on_chunk_complete
+};
+
+RestWebServer::RestWebServer(otbr::Ncp::ControllerOpenThread *aNcp):
+    mNcp(aNcp),
+    mAddress(new sockaddr_in())
+    
+{
+   
+   
+
+  
+}
+
+void RestWebServer::init(){
+   
+    mThreadHelper = mNcp->GetThreadHelper();
+    mInstance = mThreadHelper->GetInstance();
+    
+    handlerInit();
+    
+    mAddress->sin_family = AF_INET;
+    mAddress->sin_addr.s_addr = INADDR_ANY;
+    mAddress->sin_port = htons(80);
+
+    mListenFd = socket(AF_INET, SOCK_STREAM, 0);
+    otbrLog(OTBR_LOG_ERR, "socket success" );
+    
+    bind(mListenFd, (struct sockaddr *) mAddress, sizeof(sockaddr)) ;
+    otbrLog(OTBR_LOG_ERR, "bind success" );
+    
+    listen(mListenFd, 5);
+    otbrLog(OTBR_LOG_ERR, "listen success" );
+    
+    setNonBlocking(mListenFd);
```

Comment:
```suggestion
    SetNonBlocking(mListenFd);
```

Per OpenThread style guide, a function name should start with uppercase.

Reply:
Please re-check for anywhere else.